docs/book/reflection-abstract-factory.md

```diff
@@ -2,14 +2,33 @@
 
 - Since 3.2.0.
 
-Writing a factory class for each and every service that has dependencies
-can be tedious, particularly in early development as you are still sorting
-out dependencies.
+Writing a factory class for each and every service that has dependencies can be tedious, particularly in early development as you are still sorting out dependencies.
 
-laminas-servicemanager ships with `Laminas\ServiceManager\AbstractFactory\ReflectionBasedAbstractFactory`,
-which provides a reflection-based approach to instantiation, resolving
-constructor dependencies to the relevant services. The factory may be used as
-either an abstract factory, or mapped to specific service names as a factory:
+To alleviate this issue during development, laminas-servicemanager ships with the `ReflectionBasedAbstractFactory`, which provides a [reflection-based approach](https://www.php.net/manual/en/intro.reflection.php) to instantiation, resolving constructor dependencies to the relevant services. 
+The factory may be used as either an abstract factory or mapped to specific service names as a factory.
+
+TIP: Mapping services to the factory is more explicit and performant.
+
+The factory operates with the following constraints/features:
+
+- A parameter named `$config` type-hinted as an array will receive the application "config" service (i.e., the merged configuration).
+- Parameters type-hinted against array, but not named `$config`, will be injected with an empty array.
+- Scalar parameters will result in the factory raising an exception, unless a default value is present; if it is, that value will be used.
+- If a service cannot be found for a given typehint, the factory will raise an exception detailing this.
+
+WARNING: `$options` passed to the factory are ignored in all cases, as we cannot make assumptions about which argument(s) they might replace.
+
+Once your dependencies have stabilized and when performance is a requirement, we recommend writing a dedicated factory, as reflection can introduce performance overhead. 
+For example, you could use the [generate-factory-for-class console tool](console-tools.md#generate-factory-for-class) to do so.
+
+## Usage Example
+
+### Using Laminas MVC
+
+When using [Laminas MVC](https://docs.laminas.dev/mvc/):
```

Comment:
The package name for laminas-mvc is used most and is also the preferred way so far.
(If something refers to Mezzio, then the spelling with a capital initial is used: "Mezzio". The package names also follow the names used in Composer. Example: "mezzio-problem-details".)

```suggestion
### Using laminas-mvc

When using [laminas-mvc](https://docs.laminas.dev/mvc/):
```